pkg/kubelet/nodelease/controller.go

```diff
@@ -64,14 +64,10 @@ type controller struct {
 }
 
 // NewController constructs and returns a controller
-func NewController(clock clock.Clock, client clientset.Interface, holderIdentity string, leaseDurationSeconds int32, onRepeatedHeartbeatFailure func()) Controller {
-	var leaseClient coordclientset.LeaseInterface
-	if client != nil {
-		leaseClient = client.CoordinationV1().Leases(corev1.NamespaceNodeLease)
```

Comment:
In the issue description you mentioned:

> This is handy in situations where you want to use this controller outside
of Kubernetes (we specifically want to use it in virtual-kubelet).

With this change, the `NewController` constructive no longer enforces the lease client to be scoped to the `kube-node-lease` namespace and is now up to external callers of NewController to ensure the correct namespace is configured for the lease client. Do we need to be concerned about external callers passing in lease clients scoped to the wrong namespaces now?

Reply:
I think that we don't need to be concerned about it, and if people outside of the project want to use it, it's up to them to "use it correctly". We can have two constructors, like:

```go

// NewControllerWithClients constructs and returns a controller with these specific clients. The leaseClient should be built with the namespace `corev1.NamespaceNodeLease`
func NewControllerWithClients(clock clock.Clock, nodeClient corev1client.NodeInterface, leaseClient coordclientset.LeaseInterface, holderIdentity string, leaseDurationSeconds int32, onRepeatedHeartbeatFailure func()) Controller 

// NewController constructs and returns a controller. It builds the requisite clients from the clientset.
func NewController(clock clock.Clock, client clientset.Interface, holderIdentity string, leaseDurationSeconds int32, onRepeatedHeartbeatFailure func()) Controller 
```

Reply:
Seems error prone IMO since this controller implementation is strictly scoped to resources in `kube-node-lease`. We should enforce clients to be scoped to the kube-node-lease namespace if we can. Will defer to @dchen1107 @derekwaynecarr though.